toxcore/group.c

```diff
@@ -29,24 +29,80 @@
 
 #include "util.h"
 
-/* return 1 if the groupnumber is not valid.
- * return 0 if the groupnumber is valid.
- */
-static uint8_t groupnumber_not_valid(const Group_Chats *g_c, int groupnumber)
+typedef ptrdiff_t aint;
```

Comment:
If you think this optimization is absolutely necessary, I'd prefer if you used the C99 type  `int_fast32_t` intended to make this optimization, else just use `int32_t`.